packages/hardhat/contracts/periphery/ControllerHelper.sol

```diff
@@ -162,12 +177,66 @@ contract ControllerHelper is FlashControllerHelper, IERC721Receiver {
             _wPowerPerpAmountToBurn.add(_wPowerPerpAmountToBuy),
             _collateralToBuyWith.add(msg.value),
             uint8(FLASH_SOURCE.FLASH_W_BURN),
-            abi.encodePacked(_vaultId, _wPowerPerpAmountToBurn, _wPowerPerpAmountToBuy, _collateralToWithdraw, _collateralToBuyWith.add(msg.value))
+            abi.encodePacked(
+                _vaultId,
+                _wPowerPerpAmountToBurn,
+                _wPowerPerpAmountToBuy,
+                _collateralToWithdraw,
+                _collateralToBuyWith.add(msg.value)
+            )
         );
 
         emit FlashWBurn(msg.sender, _vaultId, _wPowerPerpAmountToBurn, _collateralToWithdraw, _wPowerPerpAmountToBuy);
     }
 
+    /**
+     * @notice mint WPowerPerp and LP into Uniswap v3 pool
+     * @param _vaultId vault ID
+     * @param _wPowerPerpAmount amount of WPowerPerp token to mint
+     * @param _collateralToMint collateral to use for minting
+     * @param _collateralToLP collateral to use for LPing
+     * @param _lowerTick LP lower tick
+     * @param _upperTick LP upper tick
+     */
+    function batchMintLp(
+        uint256 _vaultId,
+        uint256 _wPowerPerpAmount,
+        uint256 _collateralToMint,
+        uint256 _collateralToLP,
+        int24 _lowerTick,
+        int24 _upperTick
+    ) external payable {
+        require(msg.value == _collateralToMint.add(_collateralToLP), "Wrong ETH sent");
+
+        uint256 vaultId = IController(controller).mintWPowerPerpAmount{value: _collateralToMint}(
+            _vaultId,
+            _wPowerPerpAmount,
+            0
+        );
+        address token0 = IUniswapV3Pool(wPowerPerpPool).token0();
+        address token1 = IUniswapV3Pool(wPowerPerpPool).token1();
+
+        INonfungiblePositionManager.MintParams memory params = INonfungiblePositionManager.MintParams({
+            token0: token0,
+            token1: token1,
+            fee: IUniswapV3Pool(wPowerPerpPool).fee(),
+            tickLower: _lowerTick,
+            tickUpper: _upperTick,
+            amount0Desired: token0 == wPowerPerp ? _wPowerPerpAmount : _collateralToLP,
+            amount1Desired: token1 == wPowerPerp ? _wPowerPerpAmount : _collateralToLP,
+            amount0Min: 0,
+            amount1Min: 0,
+            recipient: msg.sender,
+            deadline: block.timestamp
+        });
+
+        INonfungiblePositionManager(nonfungiblePositionManager).mint{value: _collateralToLP}(params);
+
+        if (_vaultId == 0) IShortPowerPerp(shortPowerPerp).safeTransferFrom(address(this), msg.sender, vaultId);
+
+        emit BatchMintLp(msg.sender, _vaultId, _wPowerPerpAmount, _collateralToMint, _collateralToLP);
```

Comment:
Think we need to return unused ETH and oSQTH to the user and write tests to ensure that this is as expected.